src/client/requirements.js

```diff
@@ -0,0 +1,61 @@
+import { SECOND } from '../utils/constants';
+
+export const DEFAULTS = {
+	freshness: Number.MAX_SAFE_INTEGER,
+	timeout: 20 * SECOND,
+};
+
+/**
+ * Combines component requirements into a requirements list by endpoint.
+ * @param {Map} requirementsByComponent Key: Component, Value: requirements with endpoint/params property.
+ * @return {Object} New requirements endpoint tree.
+ */
+export function combineComponentRequirements( requirementsByComponent ) {
+	const requirements = {};
+
+	requirementsByComponent.forEach( ( requirement ) => {
+		const { endpoint, params, ...reqParams } = requirement;
+		addEndpointRequirement( requirements, reqParams, endpoint, params );
+	} );
+
+	return requirements;
+}
+
+/**
+ * Mutates the state of requirementsByEndpoint by adding a given endpoint requirement to it.
+ * @param {Object} requirementsByEndpoint Endpoint tree with requirement leaf nodes.
```

Comment:
Likewise here, should we validate inputs and/or provide defaults?

Reply:
How about this? Any functions like these get exported through the main `index.js` I'll make sure to add validation to the inputs, as that will be the "public" API and would then warrant a little defensive programming.

Reply:
Gotcha, that sounds totally fair to me.